From: Ben Hutchings <ben@decadent.org.uk>
Date: Mon, 17 Dec 2018 16:44:14 +0000
Subject: percpu_counter: Avoid ABI change in 3.16.62
Forwarded: not-needed

Commit 5f2c8b3ceecd "percpu_counter: batch size aware
__percpu_counter_compare()" changed percpu_counter_compare() from
an extern export function to a static inline function.  Change it
back.

---
--- a/include/linux/percpu_counter.h
+++ b/include/linux/percpu_counter.h
@@ -41,11 +41,7 @@ void percpu_counter_set(struct percpu_co
 void __percpu_counter_add(struct percpu_counter *fbc, s64 amount, s32 batch);
 s64 __percpu_counter_sum(struct percpu_counter *fbc);
 int __percpu_counter_compare(struct percpu_counter *fbc, s64 rhs, s32 batch);
-
-static inline int percpu_counter_compare(struct percpu_counter *fbc, s64 rhs)
-{
-	return __percpu_counter_compare(fbc, rhs, percpu_counter_batch);
-}
+int percpu_counter_compare(struct percpu_counter *fbc, s64 rhs);
 
 static inline void percpu_counter_add(struct percpu_counter *fbc, s64 amount)
 {
--- a/lib/percpu_counter.c
+++ b/lib/percpu_counter.c
@@ -216,6 +216,12 @@ int __percpu_counter_compare(struct perc
 }
 EXPORT_SYMBOL(__percpu_counter_compare);
 
+int percpu_counter_compare(struct percpu_counter *fbc, s64 rhs)
+{
+	return __percpu_counter_compare(fbc, rhs, percpu_counter_batch);
+}
+EXPORT_SYMBOL(percpu_counter_compare);
+
 static int __init percpu_counter_startup(void)
 {
 	compute_batch_value();
